src/modules/85-cv/components/HarnessServiceAndEnvironment/components/EnvironmentSelectOrCreate/EnvironmentSelectOrCreate.tsx

```diff
@@ -46,30 +49,52 @@ export function EnvironmentSelectOrCreate({
   onSelect,
   disabled,
   onNewCreated,
-  className
+  className,
+  isMultiType = false
 }: EnvironmentSelectOrCreateProps): JSX.Element {
   const { getString } = useStrings()
-
+  const { expressions } = useVariablesExpression()
   const { environmentOptions, openHarnessEnvironmentModal } = useEnvironmentSelectOrCreate({ options, onNewCreated })
-  const onSelectChange = (val: SelectOption): void => {
-    if (val.value === ADD_NEW_VALUE) {
+
+  const onSelectChange = (val: SelectOption | string): void => {
+    if (typeof val !== 'string' && val?.value === ADD_NEW_VALUE) {
       openHarnessEnvironmentModal()
     } else {
-      onSelect(val)
+      if (typeof val === 'string') {
+        onSelect({ label: val, value: val })
+      } else {
+        onSelect(val)
+      }
     }
   }
 
   return (
     <Container onClick={e => e.stopPropagation()}>
-      <Select
-        name={'environment'}
-        value={item}
-        className={className}
-        disabled={disabled}
-        items={environmentOptions}
-        inputProps={{ placeholder: getString('cv.selectOrCreateEnv') }}
-        onChange={onSelectChange}
-      />
+      {isMultiType ? (
+        <MultiTypeInput
+          name={'environmentRef'}
+          disabled={disabled}
+          selectProps={{
+            items: environmentOptions
+          }}
+          allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+          value={item}
+          style={{ width: '400px' }}
+          expressions={expressions}
+          onChange={(val: any) => onSelectChange(val)}
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9235241)
```suggestion
          onChange={(val: unknown) => onSelectChange(val)}
```